templates/powershell/build/GenerateFluxManifest.ps1

```diff
@@ -0,0 +1,120 @@
+    [Parameter(Mandatory)]
+    [string]$EnvName
+)
+
+function Add-Environment {
+    param (
+        [Parameter(Mandatory)]
+        [string]$ApiBaseUri,
+        [Parameter(Mandatory)]
+        [string]$TeamName,
+        [Parameter(Mandatory)]
+        [string]$ServiceName,
+        [Parameter(Mandatory)]
+        [string]$Name
+    )
+    begin {
+        [string]$functionName = $MyInvocation.MyCommand
+        Write-Debug "${functionName}:Entered"
+        Write-Debug "${functionName}:ApiBaseUri=$ApiBaseUri"
+        Write-Debug "${functionName}:TeamName=$TeamName"
+        Write-Debug "${functionName}:ServiceName=$ServiceName"
+        Write-Debug "${functionName}:Name=$Name"
+    }
+    process {
+        $uri = "$ApiBaseUri/FluxTeamConfig/$TeamName/services/$ServiceName/environments"
+        Write-Debug "${functionName}:Uri=$uri"
+
+        Invoke-RestMethod -Uri $uri -Method Post -Body (@($Name) | ConvertTo-Json) -ContentType "application/json"
+    }
+    end {
+        Write-Debug "${functionName}:Exited"
+    }
+}
+
+function Add-FluxConfig {
+    param (
+        [Parameter(Mandatory)]
+        [string]$ApiBaseUri,
+        [Parameter(Mandatory)]
+        [string]$TeamName,
+        [Parameter(Mandatory)]
+        [string]$ServiceName,
+        [Parameter(Mandatory)]
+        [string]$EnvName
+    )
+    begin {
+        [string]$functionName = $MyInvocation.MyCommand
+        Write-Debug "${functionName}:Entered"
+        Write-Debug "${functionName}:ApiBaseUri=$ApiBaseUri"
+        Write-Debug "${functionName}:TeamName=$TeamName"
+        Write-Debug "${functionName}:ServiceName=$ServiceName"
+        Write-Debug "${functionName}:EnvName=$EnvName"
+    }
+    process {
+        $uri = "$ApiBaseUri/FluxTeamConfig/$TeamName/generate?serviceName=$ServiceName&environment=$EnvName"
+        Write-Debug "${functionName}:Uri=$uri"
+        
+        Invoke-RestMethod -Uri $uri -Method Post
+    }
+    end {
+        Write-Debug "${functionName}:Exited"
+    }
+}
+
+
+Set-StrictMode -Version 3.0
+
+[string]$functionName = $MyInvocation.MyCommand
+[datetime]$startTime = [datetime]::UtcNow
+
+[int]$exitCode = -1
+[bool]$setHostExitCode = (Test-Path -Path ENV:TF_BUILD) -and ($ENV:TF_BUILD -eq "true")
+[bool]$enableDebug = (Test-Path -Path ENV:SYSTEM_DEBUG) -and ($ENV:SYSTEM_DEBUG -eq "true")
+
+Set-Variable -Name ErrorActionPreference -Value Continue -scope global
+Set-Variable -Name InformationPreference -Value Continue -Scope global
+
+if ($enableDebug) {
+    Set-Variable -Name VerbosePreference -Value Continue -Scope global
+    Set-Variable -Name DebugPreference -Value Continue -Scope global
+}
+
+Write-Host "${functionName} started at $($startTime.ToString('u'))"
+Write-Debug "${functionName}:TeamName=$TeamName"
+Write-Debug "${functionName}:ServiceName=$ServiceName"
+Write-Debug "${functionName}:EnvName=$EnvName"
+
+try {
+    Write-Host "Adding environment '$EnvName' to service '$ServiceName' for team '$TeamName'"
```

Comment:
Do we check if team is onboarded first? should we? and throw error if team not found?

Reply:
It will throw error if the service not found in the teams config, the service config only added from the portal.